core/src/main/java/gg/solarmc/loader/SolarPlayer.java

```diff
@@ -22,33 +22,61 @@
 import gg.solarmc.loader.data.DataKey;
 import gg.solarmc.loader.data.DataObject;
 
+import java.util.Optional;
 import java.util.UUID;
 
+/**
+ * A player whose data may be manipulated. <br>
+ * <br>
+ * Instances of {@code SolarPlayer}, whether online or offline, should not be retained
+ * in memory by API users. Only framework implementors should do so.
+ *
+ */
 public interface SolarPlayer {
 
 	/**
 	 * Gets the user ID for this player. Database schema is typically keyed by this ID.
 	 *
 	 * @return the user ID
 	 */
-	int getUserID();
+	int getUserId();
 
 	/**
 	 * Gets the minecraft UUID of this player. <br>
 	 * <br>
-	 * Note: It is more effective to utilize the ID of the player returned from {@link #getUserID()}
+	 * Note: It is more effective to utilize the ID of the player returned from {@link #getUserId()}
 	 * than to use the UUID
 	 *
 	 * @return the player's minecraft UUID
 	 */
 	UUID getMinecraftUUID();
 
+	/**
+	 * Determines whether the underlying player is online in the running JVM. <br>
+	 * <br>
+	 * It is possible for this method to return {@code false} if the user is online
```

Comment:
It is possible for this method to return false _even_ if the user is online

Reply:
I avoid unnecessary words, so I didn't think to add 'even.'

I probably should say "running server instance" rather than "running JVM."